rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRedundantDependencyVersionsTest.java

```diff
@@ -1170,6 +1229,116 @@ void removeRedundantVersionsFromPluginsManagedByParent() {
         );
     }
 
+    @Test
+    void removeRedundantManagedVersionFromPluginsManagedByParentAndKeepOtherElements() {
+        rewriteRun(
+          spec -> spec.recipe(new RemoveRedundantDependencyVersions(null, null, RemoveRedundantDependencyVersions.Comparator.GTE, null)),
+          pomXml(
+            """
+              <project>
+                  <modelVersion>4.0.0</modelVersion>
+                  <groupId>org.sample</groupId>
+                  <artifactId>sample</artifactId>
+                  <version>1.0.0</version>
+                  <parent>
+                      <groupId>org.springframework.boot</groupId>
+                      <artifactId>spring-boot-starter-parent</artifactId>
+                      <version>2.7.18</version>
+                      <relativePath/>
+                  </parent>
+                  <build>
+                      <pluginManagement>
+                          <plugins>
+                              <plugin>
+                                  <groupId>pl.project13.maven</groupId>
+                                  <artifactId>git-commit-id-plugin</artifactId>
+                                  <version>4.9.10</version>
+                                  <configuration><whatever /></configuration>
+                              </plugin>
+                          </plugins>
+                      </pluginManagement>
+                  </build>
+              </project>
+              """,
+            """
+              <project>
+                  <modelVersion>4.0.0</modelVersion>
+                  <groupId>org.sample</groupId>
+                  <artifactId>sample</artifactId>
+                  <version>1.0.0</version>
+                  <parent>
+                      <groupId>org.springframework.boot</groupId>
+                      <artifactId>spring-boot-starter-parent</artifactId>
+                      <version>2.7.18</version>
+                      <relativePath/>
+                  </parent>
+                  <build>
+                      <pluginManagement>
+                          <plugins>
+                              <plugin>
+                                  <groupId>pl.project13.maven</groupId>
+                                  <artifactId>git-commit-id-plugin</artifactId>
+                                  <configuration><whatever /></configuration>
```

Comment:
I had earlier wondered if this is allowed in Maven: to have a pluginManagement plugin without a version. I think I've seen this fail in practice, but it could have been that the circumstances were ever so slightly different.

Reply:
Let me know your thoughts on this one and then we'll merge right after.

Reply:
I added a test for non-inherited plugins (`keepPluginIfParentInheritanceDisabled()`)… and I was surprised that it succeeded out-of-the-box.

So I added another test for non-inherited within the same pom (`removeRedundantVersionsFromManagedButNonInheritedPlugins()`) and it fails: apparently [`ResolvedPom.mergePlugins()`][1] completely ignores non-inherited plugin, so the recipe does not see the managed plugin!

It’s a different issue though, so it should maybe be handled separately (might need deeper refactoring of the `mergePlugins()` logic).

Another side note: I noticed that IntelliJ reformats some parts of `RemoveRedundantDependencyVersionsTest` that I didn’t touch, so I don’t know if my setup is correct. I see there is an `.editorconfig` file there, but it does not contain much (and I see these files mostly only exist in test folders? shouldn’t there be one at the root of the repository?)

[1]: https://github.com/openrewrite/rewrite/blob/c39a0836c7073104c465ce2c592533948b8fbd3c/rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java#L702-L704

Reply:
Ah I had for now worked around this, but perhaps there's a better fix indeed in that mergePlugins; would love to hear as I'm switching between a few too many PRs in a day to fully dive in. :)

Reply:
I don’t know the codebase well enough, but I suppose a distinction could be made between plugin management and inherited plugin management. In this particular case we want to see the full plugin management for the current pom, but only the inherited one from the parent pom.

The same might be needed for non-managed plugins too…

OTOH maybe it’s good enough like this? It could possibly be more important to document what `getPlugins()`/`getPluginManagement()` return, as they aren’t very clear about inheritance.